calblueprint/siren
# Suraj/multilingual

# ✨ New in this PR ✨

## One liner: what did you do? 🚨 
flesh out screens of multilingual feature and setting overall language of app in various locations of app.
## Coverage 🙆‍♀️
_use this section to break up your task into submodules and track progress_ 
 - [x] Task 1: create settings page to handle changing user language, email, and password (handled re-authentication by requiring user to enter current password)
 - [x] Task 2: create language picking screen (first screen seen if user is not logged in)
 - [x] Task 3: carry over language from language picking screen to register screen and assign to user once registered
 - [x] Task 4: set language context of app to language chosen at language screen if not logged in. otherwise, set language context of app to current language preference of user

## How can the reviewer test your code? 👩‍💻
run `npm start` without a user logged in. should be taken to a language selector screen with buttons for each language. selecting any language should change the language of the app (text on screen should change). from there, continue on to the register screen (as you continue the language should still be changed for all subsequent screens). after successfully registering a user, check to see on firebase if they have the correct field for "language" that they had picked form the language selector screen. once logged in/registered, you can check out the settings screen by clicking the icon in the top left of the home screen. you should see text fields where you can write info to change your email, password, and language preference. try each one (and/or combos of them) and after check to see if the user's info on firebase storage changed accordingly. to see if it also worked correctly in changing user's info on firebase authentication, try logging back in with the new email/password that you had changed to. to test to make sure re-authentication is working you can log into the app and then wait a while (not exactly sure how long) and then try changing password and putting in current password and the change should still be successful. changing the language of the current user should also make the app change to the language you just inputted. additionally, you should try logging into the app, then closing and reopening it, and the app should be in the language preference that is currently associated with the logged in user.

## Afterthoughts
Once we finish actually replacing each hardcoded text block in the app with a variable that's able to be changed by the current language of the app, testing this pr will likely be much easier. 

## Any bugs you encountered or still having trouble with? 🐛


## Resources 📔
_link online resources and related PRs_ 


🧜‍♀️ cc: @shannonbonet @phoebeli23
